packages/tinacms/src/admin/components/GetCollection.tsx

```diff
@@ -66,10 +67,85 @@ export const useGetCollection = (
       }
     }
 
+    const searchCollection = async () => {
+      if ((await api.isAuthenticated()) && !folder.loading && !cancelled) {
+        try {
+          const response = (await cms.api.search.query(
+            `${search} AND _collection:${collectionName}`,
+            {
+              limit: 15,
+              cursor: after,
+            }
+          )) as {
+            results: { _id: string }[]
+            nextCursor: string
+            prevCursor: string
+          }
+          const docs = (await Promise.allSettled<
+            Promise<{ document: DocumentForm }>
+          >(
+            response.results.map((result) => {
+              const [collection, relativePath] = result._id.split(':')
+              return api.fetchDocument(collection, relativePath, false)
+            })
+          )) as {
+            status: 'fulfilled' | 'rejected'
+            value: { document: DocumentForm }
+          }[]
+          const edges = docs
+            .filter((p) => p.status === 'fulfilled' && !!p.value?.document)
+            .map((result) => ({ node: result.value.document })) as any[]
+          const { name, order } = JSON.parse(sortKey || '{}')
+          const validSortKey = collectionExtra.fields
+            ?.map((x) => x.name)
+            .includes(name)
+            ? name
+            : undefined
+          const c = await api.fetchCollection(
+            collectionName,
+            false,
+            filterArgs?.filterField ? '' : folder.fullyQualifiedName,
+            after,
+            validSortKey,
+            order,
+            filterArgs
+          )
+          setCollection({
+            format: collection.format,
+            label: collection.label,
+            name: collectionName,
+            templates: collection.templates,
+            documents: {
+              pageInfo: {
+                hasNextPage: !!response.nextCursor,
+                hasPreviousPage: !!response.prevCursor,
+                startCursor: '',
+                endCursor: response.nextCursor || '',
+              },
+              edges,
+            },
+          })
+        } catch (error) {
+          cms.alerts.error(
+            `[${error.name}] GetCollection failed: ${error.message}`
+          )
+          console.error(error)
+          setCollection(undefined)
+          setError(error)
+        }
+
+        setLoading(false)
+      }
+    }
+
     if (cancelled) return
 
     setLoading(true)
-    fetchCollection()
+    if (search) {
+      searchCollection()
```

Comment:
Not sure I understand what we're doing here. Won't this keep looping

Reply:
I don't _think_ it will loop - its not changing any of the useEffect dependencies

Reply:
Oops! My bad, thought we were calling searchCollection recursively